### PR TITLE
[JB][SUPP-317] add audit event for Payment Plan details before submis…

### PR DESCRIPTION
**Pull request by barnesjake**
…sion

### CODE REVIEW

File: app/controllers/PaymentScheduleController.scala [hunk 1]
```diff
@@ -65,9 +66,13 @@ class PaymentScheduleController @Inject() (
...
 
       case j: Journey.AfterSelectedPaymentPlan =>
-        journeyService.updateHasCheckedPaymentPlan(j.journeyId)
-          .map(_ => Redirect(routes.BankDetailsController.typeOfAccount))
-
+        for {
+          _ <- journeyService.updateHasCheckedPaymentPlan(j.journeyId)
+          _ = j match {
+            case j1: Journey.Stages.ChosenPaymentPlan => auditService.auditPaymentPlanBeforeSubmission(j1)
+            case _                                    => JourneyLogger.info("Nothing to audit, so I won't")
```
Comment (AndyHWChung) [inline app/controllers/PaymentScheduleController.scala:73]: probably don't need this log at info level - maybe debug with a mention of what stage it is at so we've got the info if we need it?

File: app/controllers/PaymentScheduleController.scala [hunk 2]
```diff
@@ -65,9 +66,13 @@ class PaymentScheduleController @Inject() (
         logErrorAndRouteToDefaultPageF(j)
 
       case j: Journey.AfterSelectedPaymentPlan =>
-        journeyService.updateHasCheckedPaymentPlan(j.journeyId)
-          .map(_ => Redirect(routes.BankDetailsController.typeOfAccount))
-
+        for {
+          _ <- journeyService.updateHasCheckedPaymentPlan(j.journeyId)
+          _ = j match {
```
Comment (AndyHWChung) [inline app/controllers/PaymentScheduleController.scala:71]: small thing here but could we audit first and then update mongo? With audit events it's best practice to stick the event as close as possible to the thing that triggered it. Also, in the very unlikely scenario that the BE update failed then we still have a chance of having the audit event there.

Comment (barnesjake) [reply app/controllers/PaymentScheduleController.scala:71]: Yeah sure thing

File: app/controllers/PaymentScheduleController.scala (hunk 1 above)
Comment (barnesjake) [reply app/controllers/PaymentScheduleController.scala:73]: Will do

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**